lib/foreman/renderer.rb

```diff
@@ -233,6 +252,10 @@ def load_template_vars
       @template_url = params['url']
     end
 
+    def global_setting(name, default = nil)
+      SETTINGS[name.to_sym] || default
```

Comment:
I believe this was rejected in past because setting contain sensitive information, such as root password for newly provisioned hosts. I suppose we'd need to add a whitelist here. Also the || won't work for settings that has value `false`, it would always return default value. If you're going to fix this, I'd also suggest renaming `global_setting` to `setting`, there's no other setting...

Reply:
Also note that settings page is only available for admins, there's no permission for this. Making it available through template give ability to everyone who can render (preview) template, to see values.

Reply:
So how about renaming it to "setting", creating whitelist, fixing default value and allowing override via host params, so people can still override this per host/hostgroup.

Reply:
Or maybe is cleaner just to use host/hostgroup parameter for this. That would drop this complexity. But to be honest I really like using settings in templates.

Reply:
If that is for admins only, then I think we can drop the host/hg params. Whitelist, default value and renaming is still required I think. I think the whitelist should contain nearly all settings and (sorry to say that), there should be a plugin DSL for extending it. I suppose discovery plugin would require it.